File: app/src/main/java/com/sprotte/geolocator/demo/java/LocationTrackerWorker.java
package com.sprotte.geolocator.demo.java;

import android.content.Context;

import androidx.annotation.NonNull;

import com.google.android.gms.location.LocationResult;
import com.sprotte.geolocator.demo.misc.UtilsKt;
import com.sprotte.geolocator.geofencer.models.LocationTrackerUpdateModule;

public class LocationTrackerWorker extends LocationTrackerUpdateModule {

    private Context context;

    LocationTrackerWorker(Context context) {
        super(context);
        this.context = context;
    }

    @Override
    public void onLocationResult(@NonNull LocationResult locationResult) {
        UtilsKt.sendNotification(
                context,
                "Location Update",
                locationResult.toString()
        );
    }
}

File: app/src/main/java/com/sprotte/geolocator/demo/kotlin/LocationTrackerWorker.kt
package com.sprotte.geolocator.demo.kotlin

import android.content.Context
import android.util.Log
import androidx.core.content.edit
import com.google.android.gms.location.LocationResult
import com.sprotte.geolocator.demo.misc.toJson
import com.sprotte.geolocator.geofencer.models.LocationTrackerUpdateModule
import timber.log.Timber

class LocationTrackerWorker(context: Context) : LocationTrackerUpdateModule(context) {

    override fun onLocationResult(locationResult: LocationResult) {

        Timber.v("locationResult=$locationResult")

        sharedPreferences?.edit {
            putString(USER_LOCATION_KEY, locationResult.toJson())
        }
    }

    companion object {
        const val USER_LOCATION_KEY = "user_location"
    }
}
